src/hera/workflows/_mixins.py

```diff
@@ -74,6 +74,15 @@
 EnvT = Optional[Union[Union[_BaseEnv, EnvVar], List[Union[_BaseEnv, EnvVar]]]]
 EnvFromT = Optional[Union[Union[_BaseEnvFrom, EnvFromSource], List[Union[_BaseEnvFrom, EnvFromSource]]]]
 TContext = TypeVar("TContext", bound="ContextMixin")
+THookable = TypeVar("THookable", bound="HookMixin")
+
+
+class HookMixin(BaseMixin):
+    def _dispatch_hooks(self: THookable) -> THookable:
+        output = self
+        for hook in global_config._get_pre_build_hooks(output):
+            output = hook(output)
```

Comment:
Is it necessary for the hook to return the output? I wonder if there are instances in which users will forget to actually return the object, and will result in a `None` type _somewhere_. I think it might be easier for users to design a hook that takes an object and modifies the reference/applies changes in place